lightning/src/util/chacha20poly1305rfc.rs

```diff
@@ -132,14 +132,12 @@ impl<'a, W: Writer> Writer for ChaChaPolyWriter<'a, W> {
 	// complete.
 	fn write_all(&mut self, src: &[u8]) -> Result<(), io::Error> {
 		let mut src_idx = 0;
-		let mut write_buffer = [0; 8192];
-		let mut bytes_written = (&mut write_buffer[..]).write(&src[src_idx..]).expect("In-memory writes can't fail");
-		while bytes_written != 0 {
+		while src.len() - src_idx > 0 {
```

Comment:
Could avoid the subtraction by doing `src.len() > src_idx` or maybe more naturally as `src_idx < src.len()`.